From: Arnaldo Carvalho de Melo <acme@ghostprotocols.net>
Subject: Annotate core code that should not be traced

Mark with "notrace" functions in core code that should not be
traced.  The "notrace" attribute will prevent gcc from adding
a call to mcount on the annotated funtions.

Signed-off-by: Arnaldo Carvalho de Melo <acme@ghostprotocols.net>
Signed-off-by: Steven Rostedt <srostedt@redhat.com>

---
 lib/smp_processor_id.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.24-rt1/lib/smp_processor_id.c
===================================================================
--- linux-2.6.24-rt1.orig/lib/smp_processor_id.c	2008-01-25 15:06:51.000000000 -0500
+++ linux-2.6.24-rt1/lib/smp_processor_id.c	2008-01-25 15:06:52.000000000 -0500
@@ -7,7 +7,7 @@
 #include <linux/kallsyms.h>
 #include <linux/sched.h>
 
-unsigned int debug_smp_processor_id(void)
+notrace unsigned int debug_smp_processor_id(void)
 {
 	unsigned long preempt_count = preempt_count();
 	int this_cpu = raw_smp_processor_id();
